controller/src/main/java/org/jboss/as/controller/CapabilityRegistry.java

```diff
@@ -940,16 +940,33 @@ private static Set<RuntimeRequirementRegistration> findInconsistent(Map<Capabili
 
     private RuntimeCapabilityRegistration getCapabilityRegistration(String capabilityName, CapabilityScope capabilityScope) {
         SatisfactoryCapability satisfactoryCapability = findSatisfactoryCapability(capabilityName, capabilityScope, false);
+
         if (satisfactoryCapability == null) {
             if (forServer) {
-                throw ControllerLogger.MGMT_OP_LOGGER.unknownCapability(capabilityName);
+                if (isInCapability(capabilityName)) {
+                    throw ControllerLogger.MGMT_OP_LOGGER.cannotGetCapability(capabilityName);
+                } else {
+                    throw ControllerLogger.MGMT_OP_LOGGER.missingCapability(capabilityName);
+                }
```

Comment:
This if (forServer) block should just be the existing code, unchanged.

For the 'forServer' case, findSatisfactoryCapability is just a simple lookup. There's no difference between 'capability exists' and 'capability is satisfactory'. The existing message covers this.